test_conformance/relationals/test_comparisons_fp.cpp

```diff
@@ -0,0 +1,662 @@
+"{\n"
+"    int  tid = get_global_id(0);\n"
+"    destValues[tid] = %s( sourceA[tid], sourceB[tid] );\n"
+"    destValuesB[tid] = (sourceA[tid] < sourceB[tid]) | (sourceA[tid] > sourceB[tid]);\n"
+"}\n"};
+
+const char* equivTestKerPat_3[] = {
+extension,
+"__kernel void sample_test(__global ", ftype_vec, " *sourceA, __global ", ftype_vec,
+" *sourceB, __global ", itype_vec, " *destValues, __global ", itype_vec, " *destValuesB)\n"
+"{\n"
+"    int  tid = get_global_id(0);\n"
+"    ",ftype_vec," sampA = vload3(tid, (__global ",ftype," *)sourceA);\n"
+"    ",ftype_vec," sampB = vload3(tid, (__global ",ftype," *)sourceB);\n"
+"    vstore3(%s( sampA, sampB ), tid, (__global ",itype," *)destValues);\n"
+"    vstore3(( sampA %s sampB ), tid, (__global ",itype," *)destValuesB);\n"
+"}\n"};
+
+const char* equivTestKerPatLessGreater_3[] = {
+extension,
+"__kernel void sample_test(__global ", ftype_vec, " *sourceA, __global ", ftype_vec,
+" *sourceB, __global ", itype_vec, " *destValues, __global ", itype_vec, " *destValuesB)\n"
+"{\n"
+"    int  tid = get_global_id(0);\n"
+"    ", ftype_vec, " sampA = vload3(tid, (__global ", ftype, " *)sourceA);\n"
+"    ", ftype_vec, " sampB = vload3(tid, (__global ", ftype, " *)sourceB);\n"
+"    vstore3(%s( sampA, sampB ), tid, (__global ", itype, " *)destValues);\n"
+"    vstore3(( sampA < sampB ) | (sampA > sampB), tid, (__global ", itype, " *)destValuesB);\n"
+"}\n"
+};
+// clang-format on
+//--------------------------------------------------------------------------/
+
+std::string concat_kernel(const char* sstr[], int num)
+{
+    std::string res;
+    for (int i = 0; i < num; i++) res += std::string(sstr[i]);
+    return res;
+}
+
+//--------------------------------------------------------------------------/
+
+template <typename... Args>
+std::string string_format(const std::string& format, Args... args)
+{
+    int size_s = std::snprintf(nullptr, 0, format.c_str(), args...)
+        + 1; // Extra space for '\0'
+    if (size_s <= 0)
+    {
+        throw std::runtime_error("Error during formatting.");
+    }
+    auto size = static_cast<size_t>(size_s);
+    std::unique_ptr<char[]> buf(new char[size]);
+    std::snprintf(buf.get(), size, format.c_str(), args...);
+    return std::string(buf.get(),
+                       buf.get() + size - 1); // We don't want the '\0' inside
+}
+
+//--------------------------------------------------------------------------
+
+template <typename T, typename F> bool verify(const T& A, const T& B)
+{
+    return F()(A, B);
+}
+
+//--------------------------------------------------------------------------
+
+RelationalsFPTest::RelationalsFPTest(cl_device_id device, cl_context context,
+                                     cl_command_queue queue, const char* fn,
+                                     const char* op)
+    : device(device), context(context), queue(queue), fnName(fn), opName(op)
```

Comment:
The initialization order should follow the class member declaration order, so:
```suggestion
    : context(context), device(device), queue(queue), fnName(fn), opName(op)
```
For consistency you could perhaps make `context` the first constructor argument.

Reply:
Done, thanks!